Seems to catch two issues.

Hosted: (Already fixed)
```
modm/src/modm/architecture/detect.hpp:228:3: error: #error "Couldn't determine target CPU!"
228 | # error "Couldn't determine target CPU!"
```

ARM Cortex-M:
```
Linking········ build/release/blink.elf
scons: building terminated because of errors.`_sbrk_r' referenced in section `.text' of /usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/lib/thumb/v7-m/nofp/libc_nano.a(lib_a-nano-mallocr.o): defined in discarded section `.Heap_is_not_implemented!__Please_include_the__modm:platform:heap__module_in_your_project!' of build/release/modm/libmodm.a(no_heap.o)
`_sbrk_r' referenced in section `.text' of /usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/lib/thumb/v7-m/nofp/libc_nano.a(lib_a-nano-mallocr.o): defined in discarded section `.Heap_is_not_implemented!__Please_include_the__modm:platform:heap__module_in_your_project!' of build/release/modm/libmodm.a(no_heap.o)
`_sbrk_r' referenced in section `.text' of /usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/lib/thumb/v7-m/nofp/libc_nano.a(lib_a-nano-mallocr.o): defined in discarded section `.Heap_is_not_implemented!__Please_include_the__modm:platform:heap__module_in_your_project!' of build/release/modm/libmodm.a(no_heap.o)
collect2: error: ld returned 1 exit status
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,0 +1,22 @@
+os: linux
+dist: focal  # Ubuntu 20.04LTS
+arch: arm64
+#arch: arm64-graviton2
+
+language: minimal
+
+before_install:
+  - sudo apt-get update
+  - sudo apt-get install -y python3 python3-pip scons cmake doxygen gcc build-essential libboost-all-dev libwiringpi-dev
+  #- sudo apt-get install -y gcc-arm-none-eabi
+  - wget -qO- https://armkeil.blob.core.windows.net/developer/Files/downloads/gnu-rm/9-2020q2/gcc-arm-none-eabi-9-2020-q2-update-aarch64-linux.tar.bz2 | tar xj -C /opt
+  - export PATH="/opt/gcc-arm-none-eabi-9-2020-q2-update/bin:$PATH"
+  - pip3 install modm
+  - export PATH="~/.local/bin:$PATH"
+  - uname -a
+
+script:
+  - (cd test && make run-hosted-linux)
+  - (cd examples && ../tools/scripts/examples_compile.py linux)
+  - (cd examples && ../tools/scripts/examples_compile.py stm32f1_discovery nucleo_f103rb olimexino_stm32 stm32f103c8t6_blue_pill stm32f103c8t6_black_pill)
+  - (cd examples && ../tools/scripts/examples_compile.py rpi)

--- a/src/modm/architecture/detect.hpp
+++ b/src/modm/architecture/detect.hpp
@@ -200,7 +200,7 @@
 #	endif
 #endif
 
-#if defined __arm__ || defined __ARM_EABI__
+#if defined __arm__ || defined __aarch64__
 #	define MODM_CPU_ARM	1
 #	define MODM_ALIGNMENT	4
 #	if defined __ARM_ARCH_4T__
@@ -208,19 +208,22 @@
 #		define MODM_CPU_STRING		"ARM7TDMI"
 #	elif defined __ARM_ARCH_6SM__ || defined __ARM_ARCH_6M__
 #		define MODM_CPU_CORTEX_M0	1
-#		define MODM_CPU_STRING		"Cortex-M0"
+#		define MODM_CPU_STRING		"ARM Cortex-M0"
 #	elif defined __ARM_ARCH_7M__
 #		define MODM_CPU_CORTEX_M3	1
-#		define MODM_CPU_STRING		"Cortex-M3"
+#		define MODM_CPU_STRING		"ARM Cortex-M3"
 #	elif defined __ARM_ARCH_7EM__
 #		define MODM_CPU_CORTEX_M4	1
-#		define MODM_CPU_STRING		"Cortex-M4"
+#		define MODM_CPU_STRING		"ARM Cortex-M4"
 #	elif defined __ARM_ARCH_6__
 #		define MODM_CPU_CORTEX_A6	1
-#		define MODM_CPU_STRING		"Cortex-A6"
+#		define MODM_CPU_STRING		"ARM Cortex-A6"
 #	elif defined __ARM_ARCH_7A__
 #		define MODM_CPU_CORTEX_A7	1
-#		define MODM_CPU_STRING		"Cortex-A7"
+#		define MODM_CPU_STRING		"ARM Cortex-A7"
+#	elif defined __ARM_ARCH_ISA_A64
+#		define MODM_CPU_AARCH64	1
+#		define MODM_CPU_STRING		"ARM AArch64"
 #	endif
 #endif
 

--- a/src/modm/math/utils/bit_operation.hpp
+++ b/src/modm/math/utils/bit_operation.hpp
@@ -107,9 +107,16 @@ namespace modm
 	modm_always_inline uint32_t
 	swap(uint32_t n)
 	{
-#ifdef MODM_CPU_ARM
+#if defined(MODM_CPU_ARM) && !defined(__aarch64__)
 		asm volatile(
 			"rev %0,%0"		"\n\t"
+			 : "=r" (n)
+			 : "0" (n)
+		);
+		return n;
+#elif defined(MODM_CPU_ARM) && defined(__aarch64__)
+		asm volatile(
+			"rev32 %0,%0"	"\n\t"
 			 : "=r" (n)
 			 : "0" (n)
 		);
@@ -187,9 +194,18 @@ namespace modm
 	inline uint32_t
 	bitReverse(uint32_t n)
 	{
-#if defined(MODM_CPU_ARM) && !defined(MODM_CPU_CORTEX_M0)
+#if defined(MODM_CPU_ARM) && !defined(MODM_CPU_CORTEX_M0) && !defined(__aarch64__)
 		asm volatile(
 			"rbit %0,%0"	"\n\t"
+			 : "=r" (n)
+			 : "0" (n)
+		);
+		return n;
+#elif defined(__aarch64__)
+		asm volatile(
+			"rbit %0,%0"	"\n\t"
+			"rev32 %0,%0"	"\n\t"
+			"rev %0,%0"		"\n\t"
 			 : "=r" (n)
 			 : "0" (n)
 		);

